spring-web/src/test/java/org/springframework/http/RequestEntityTests.java

```diff
@@ -79,6 +81,16 @@ public void uriVariablesExpansion() throws URISyntaxException {
 		assertThat(entity.getUrl()).isEqualTo(expected);
 	}
 
+	@Test
+	public void uriExpansion() throws URISyntaxException{
+		String template = "/foo?bar={bar}";
+		String bar = "foo";
+		final RequestEntity<Void> entity = RequestEntity.get(template).uriVariables(bar).build();
+
+		assertThat(entity.getUrl()).isEqualTo(new URI("/foo?bar=foo"));
+	}
+
```

Comment:
This is lacking a test to demonstrate expanding with a `UriTemplateHandler`, e.g. one that has a baseUrl. That was actually the original motivation for the change.